ci/scripts/install_python.sh

```diff
@@ -47,17 +47,11 @@ full_version=${versions[$2]}
 if [ $platform = "macOS" ]; then
     echo "Downloading Python installer..."
 
-    if [ "$version" = "3.13" ] || [ "$version" = "3.13t" ];
+    if [ "$(uname -m)" = "x86_64" ] && [ "$version" = "3.9" ];
```

Comment:
LGTM, can you also update https://github.com/apache/arrow/pull/44375/files#diff-404e24b955d9320a7b3db90149855e38c86133f626f8b555002212293e79ad80L31 to:
```
[3.12]=3.12.7
```
This is the latest with packages (`.pkg`) on https://www.python.org/ftp/python/